### ISSUE TITLE
Training loss and validation loss are too low

### ISSUE THREAD

**Issue by HatemMS2012** [codekansas/keras-language-modeling#35]
Thanks for the implementation and the tutorial.

I have a general question regarding the training and the validation loss. 
I created a model which is inspired by your work here to do prediction on knowledge base triples. 
My model uses hinge loss and cosine similarity between positive and negative training instance:

loss = max(0, margin - sim(pos_triple) + sim(neg_triple))

When I train the model,  I getting too small values for the training and the validation losses from the first epoch and after few epoch **the validation loss is becoming zero**. 

Did you experienced similar behavior in your experiments? Could anybody  provide an explanation?

Best,

**Comment by codekansas** [comment]
I believe this is normal. The validation loss going to zero does not mean your model is perfect, since the positive and negative examples might not be the most challenging ones that the model will face. In fact, in the actual paper they trained for significantly more epochs after the validation loss was sometimes zero. An interesting challenge might be to figure out how to make sure the examples presented are the most challenging ones.

**Comment by HatemMS2012** [comment]
Thanks for the answer!
